src/Server/TextDocument.php

```diff
@@ -124,4 +126,25 @@ private function updateAst(string $uri, string $content)
             $this->asts[$uri] = $stmts;
         }
     }
+
+    /**
+     * The document formatting request is sent from the server to the client to format a whole document.
+     *
+     * @param TextDocumentIdentifier $textDocument The document to format
+     * @param FormattingOptions $options The format options
+     * @return TextEdit[]
+     */
+    public function formatting(TextDocumentIdentifier $textDocument, FormattingOptions $options)
+    {
+        $nodes = $this->asts[$textDocument->uri];
+        if (empty($nodes)){
+            return [];
+        }
+        $prettyPrinter = new PrettyPrinterStandard();
+        $edit = new TextEdit();
+        $edit->range = new Range(new Position(0, 0), new Position(PHP_INT_MAX, PHP_INT_MAX));
```

Comment:
I think it's okay as-is. Other options would be to use `$nodes[count($nodes) - 1]->getAttribute('fileEndPos')` or introduce a new AST object that holds the statements but also the raw file content.